node/service/src/relay_chain_selection.rs

```diff
@@ -417,7 +417,7 @@ where
 
 		let lag = if cfg!(feature = "disputes") {
 			// Prevent sending flawed data to the dispute-coordinator.
-			if Some(subchain_block_descriptions.len() as _) !=
+			if Some(subchain_block_descriptions.len() as u32 - 1) !=
 				subchain_number.checked_sub(target_number)
```

Comment:
while equivalent, I would suggest to use the same calculation as in [approval voting](https://github.com/paritytech/polkadot/blob/b9750b6fd4305d0d372fb0523e69ec192a66bd5c/node/core/approval-voting/src/lib.rs#L1198). So instead of doing the -1, I would change the `checked_sub(target_number)` to `checked_sub(target_number + 1)`.